services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3CrtAsyncHttpClient.java

```diff
@@ -76,17 +79,20 @@ private S3CrtAsyncHttpClient(Builder builder) {
     }
 
     @Override
-    public CompletableFuture<Void> execute(AsyncExecuteRequest request) {
+    public CompletableFuture<Void> execute(AsyncExecuteRequest asyncRequest) {
         CompletableFuture<Void> executeFuture = new CompletableFuture<>();
-        HttpRequest httpRequest = toCrtRequest(request);
-        S3CrtResponseHandlerAdapter responseHandler = new S3CrtResponseHandlerAdapter(executeFuture, request.responseHandler());
+        URI uri = asyncRequest.request().getUri();
+        HttpRequest httpRequest = toCrtRequest(uri, asyncRequest);
+        S3CrtResponseHandlerAdapter responseHandler =
+            new S3CrtResponseHandlerAdapter(executeFuture, asyncRequest.responseHandler());
 
-        S3MetaRequestOptions.MetaRequestType requestType = requestType(request);
+        S3MetaRequestOptions.MetaRequestType requestType = requestType(asyncRequest);
 
         S3MetaRequestOptions requestOptions = new S3MetaRequestOptions()
             .withHttpRequest(httpRequest)
             .withMetaRequestType(requestType)
-            .withResponseHandler(responseHandler);
+            .withResponseHandler(responseHandler)
+            .withEndpoint(s3NativeClientConfiguration.endpointOverride());
```

Comment:
Do we need to set it on `S3MetaRequestOptions` if we already have it on `S3ClientOptions`?

Reply:
I'm just following the pattern established in their unit test here: https://github.com/awslabs/aws-crt-java/blob/bd25b0438737e3fb90c257ea18a2c5be026274ed/src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java#L213.